tests/src/App/MappedName.cpp

```diff
@@ -833,6 +833,125 @@ TEST(MappedName, startsWith)
     EXPECT_EQ(mappedName.startsWith("WASD"), false);
 }
 
+TEST(MappedName, findTagInElementNameHexPositiveIndexNonrecursive)
+{
+    // The non-recursive version will find just the last tag, prefixed by the POSTFIX_TAG (";:H").
+    // It consists of a tag (stored in hexadecimal) and a length (also stored in hex), separated by
+    // a colon. In this example, we expect the get ;:H1b:10,F as our element, which is a tag of
+    // 0x1b and an indicated length of 0x10 (16 bytes). So the output length is the position of the
+    // tag (36) minus the indicated length, giving 20.
+
+    // Arrange
+    Data::MappedName mappedName("#94;:G0;XTR;:H19:8,F;:H1a,F;BND:-1:0;:H1b:10,F");
+    long tagOutput {0};
+    int lenOutput {0};
+    std::string postfix;
+    char type {0};
+
+    // Act
+    int result =
+        mappedName.findTagInElementName(&tagOutput, &lenOutput, &postfix, &type, false, false);
+
+    // Assert
+    EXPECT_EQ(result, 36);     // The location of the tag
+    EXPECT_EQ(tagOutput, 0x1b);// The tag
+    EXPECT_EQ(lenOutput, 20);  // The calculated length based on the tag's length parameter
+    EXPECT_EQ(postfix, ";:H1b:10,F");
+    EXPECT_EQ(type, 'F');// F=Face
+}
+
+TEST(MappedName, findTagInElementNameDecPositiveIndexNonrecursive)
+{
+    // Test backwards compatibility with the older style decimal tag storage.
```

Comment:
Question about this: freecad doesn't have toponaming yet, what are we being backwards compatible with?